src/noise_filtering.cpp

```diff
@@ -232,37 +232,38 @@ void remove_leaves(std::shared_ptr<pangenome::Graph> pangraph,
                 std::cout << "remove from read " << r << ": ";
-                for (const auto &n : pangraph->reads[r]->nodes) {
-                    std::cout << n->node_id << " ";
+                for (const auto &n : pangraph->reads[r]->get_nodes()) {
+                    std::cout << n.lock()->node_id << " ";
                 }
                 std::cout << std::endl;
-                if (pangraph->reads[r]->nodes.size() == dbg.size) {
+                if (pangraph->reads[r]->get_nodes().size() == dbg.size) {
                     std::cout << "remove read";
                     pangraph->remove_read(r);
                     std::cout << " done" << std::endl;
                 } else {
                     std::cout << "remove from read ";
                     pos = pangraph->reads[r]->find_position(node_ids, node_orients);
                     std::cout << " pos " << pos.first << " " << pos.second;
-                    assert(pos.first == 0 or pos.first + node_ids.size() == pangraph->reads[r]->nodes.size());
+                    assert(pos.first == 0 or pos.first + node_ids.size() == pangraph->reads[r]->get_nodes().size());
                     if (pos.first == 0) {
-                        node = pangraph->reads[r]->nodes[0];
-                        pangraph->reads[r]->remove_node(pangraph->reads[r]->nodes.begin());
-                        node->remove_read(pangraph->reads[r]);
-                    } else if (pos.first + node_ids.size() == pangraph->reads[r]->nodes.size()) {
-                        node = pangraph->reads[r]->nodes.back();
-                        pangraph->reads[r]->remove_node(--pangraph->reads[r]->nodes.end());
-                        node->remove_read(pangraph->reads[r]);
+                        node = pangraph->reads[r]->get_nodes()[0];
+                        pangraph->reads[r]->remove_node_with_iterator(pangraph->reads[r]->get_nodes().begin());
+                        node.lock()->remove_read(pangraph->reads[r]);
+                    } else if (pos.first + node_ids.size() == pangraph->reads[r]->get_nodes().size()) {
+                        node = pangraph->reads[r]->get_nodes().back();
+                        pangraph->reads[r]->remove_all_nodes_with_this_id((--pangraph->reads[r]->get_nodes().end())->lock()->node_id);
+                        node.lock()->remove_read(pangraph->reads[r]);
                     }
                     std::cout << "read is now " << r << ": ";
-                    for (const auto &n : pangraph->reads[r]->nodes) {
-                        std::cout << n->node_id << " ";
+                    for (const auto &n : pangraph->reads[r]->get_nodes()) {
+                        std::cout << n.lock()->node_id << " ";
                     }
                     std::cout << "done" << std::endl;
                 }
             }
-            if (node and node->covg == 0) {
-                pangraph->remove_node(node);
+            auto nodeSharedPtrFromWeakPtr = node.lock();
```

Comment:
snake_case

Reply:
Done in https://github.com/leoisl/pandora/commit/5b329a78d89eaa0f85592258a8fb378677f71587